doc/hardware/peripherals/sensor/temp_polling.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/sensor.h>
+
+#define TEMP_CHANNEL {SENSOR_CHAN_AMBIENT_TEMP, 0}
+
+const struct device *const temp0 = DEVICE_DT_GET(DT_ALIAS(temp0));
+
+SENSOR_DT_READ_IODEV(temp_iodev, DT_ALIAS(temp0), {TEMP_CHANNEL});
+RTIO_DEFINE(temp_ctx, 1, 1);
+
+int main(void)
+{
+	int rc;
+	uint8_t buf[8];
+	uint32_t temp_frame_iter = 0;
+	struct sensor_q31_data temp_data = {0};
+	struct sensor_decoder_context temp_decoder = SENSOR_DECODER_CONTEXT_INIT(
+		SENSOR_DECODER_DT_GET(DT_ALIAS(dht0)), buf, SENSOR_CHAN_AMBIENT_TEMP, 0);
```

Comment:
maybe worth having a #define for the alias to avoid this

```suggestion
		SENSOR_DECODER_DT_GET(DT_ALIAS(temp0)), buf, SENSOR_CHAN_AMBIENT_TEMP, 0);
```